examples/gke-ha-setup-using-spot-vms/k8s-deployment.tf

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "kubernetes_namespace" "my_app_ns" {
+  metadata {
+    name = var.app_name
+
+    annotations = {
+      name = "${var.app_name}-namespace"
+    }
+  }
+}
+
+resource "kubernetes_deployment" "my_app" {
```

Comment:
This is pretty hard to follow. What do you think about using [kubernetes_manifest](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest), a regular YAML manifest, and [`yamldecode`](https://www.terraform.io/language/functions/yamldecode) to load it

Reply:
agreed, it would be prettier. However, it seems that using kubernetes_manifest requires an up and running Kubernetes cluster, which means the script would have to be split up in two steps:
https://github.com/hashicorp/terraform-provider-kubernetes-alpha/issues/199#issuecomment-832614387